zanata-war/src/main/java/org/zanata/util/DeltaSpikeWindowIdParam.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2016, Red Hat, Inc. and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.zanata.util;
+
+import org.apache.deltaspike.core.spi.scope.window.WindowContext;
+
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.Produces;
+import javax.inject.Named;
+
+/**
+ * @author Sean Flanigan <a href="mailto:sflaniga@redhat.com">sflaniga@redhat.com</a>
+ */
+public class DeltaSpikeWindowIdParam {
+    /**
+     * Returns a query string which, when appended to a URL without a query
+     * string, will preserve the DeltaSpike window id (dswid) of the current
+     * request, if any. eg "?dswid=1234". The string will be empty if there
+     * is no current window context.
+     * NB: This method assumes that the URL does not contain '?' yet.
+     * @param windowContext
+     * @return
+     */
+    @Dependent
+    @Named("dswidQuery")
+    @Produces
+    String getDswidQuery(WindowContext windowContext) {
```

Comment:
I don't see a drawback in injecting the values, as opposed to injecting a
utility class and calling it. (And without injection, a static utility
class could be a pain for testing.)

On 1 March 2016 at 10:21, Carlos Munoz <notifications@github.com> wrote:

> In zanata-war/src/main/java/org/zanata/util/DeltaSpikeWindowIdParam.java
> <https://github.com/zanata/zanata-server/pull/1072#discussion_r54503752>:
>
> > + * @author Sean Flanigan <a href="mailto:sflaniga@redhat.com">sflaniga@redhat.com</a>
> > + */
> > +public class DeltaSpikeWindowIdParam {
> > +    /**
> > +     * Returns a query string which, when appended to a URL without a query
> > +     * string, will preserve the DeltaSpike window id (dswid) of the current
> > +     * request, if any. eg "?dswid=1234". The string will be empty if there
> > +     * is no current window context.
> > +     * NB: This method assumes that the URL does not contain '?' yet.
> > +     * @param windowContext
> > +     * @return
> > +     */
> > +    @Dependent
> > +    @Named("dswidQuery")
> > +    @Produces
> > +    String getDswidQuery(WindowContext windowContext) {
>
> Do we need actual producers for these two Strings? Since they are
> dependent, isn't it easier to just call a utility class when they are
> needed?
> I guess it does decouple the generation of these strings though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/1072/files#r54503752>.
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat